### ISSUE TITLE
project plans? 

### ISSUE THREAD

**Issue by abathur** [etke/ief#1]
Hi @etke

I've been working on a tool that identifies bare references to executables in shell scripts and resolves them to absolute paths. I stumbled on `ief` while doing some early research on the sub-problem of how feasible it is to identify executables that _might_ execute their arguments in order to flag them for manual triage.

I'm focused on binaries first, but I'll need to address the same issue for at least a few scripting languages. My tool is in Python, so I'm not sure yet if it'll ultimately make sense (performance wise) to bulk-check every executable on the user-specified PATH in one call, or check them individually once they're actually known to be used in the script and have arguments.

Since I've caught myself thinking about forking ief to add a few features, I just wanted to say hey and get a sense of ~vision questions like whether you see the tool as complete or what your broader goals are.

**Comment by etke** [comment]
A little late... but fork away! Project was just created to fill a particular need at one point in time and eventually to switch to a native rust binaryninja plugin (which is now done) while keeping the command line utility as is.

**Comment by abathur** [comment]
No worries--thanks for the response. I ended up using an approach based on YARA, at least for now. For posterity (maybe just for a future version of myself): 

I'm a bit of a goldfish, but I looked over the [~exploratory code](https://github.com/abathur/commandeer/blob/44951a66c4f0501f213991b39bc1db89d8fabbba/static_exec.nix#L169-L176) I was using to compare an ief-based approach to some others and suspect the main thing I had in mind was being able to search for multiple symbols in a single invocation.
